test/parallel/test-fs-watch-recursive.js

```diff
@@ -22,6 +22,37 @@ const tmpdir = require('../common/tmpdir');
 const testDir = tmpdir.path;
 tmpdir.refresh();
 
+(async () => {
+  // Watch a folder and update an already existing file in it.
+
+  const rootDirectory = fs.mkdtempSync(testDir + path.sep);
+  const testDirectory = path.join(rootDirectory, 'test-0');
+  fs.mkdirSync(testDirectory);
+
+  const testFile = path.join(testDirectory, 'file-1.txt');
+  fs.writeFileSync(testFile, 'hello');
+
+  const watcher = fs.watch(testDirectory, { recursive: true });
+  let watcherClosed = false;
+  watcher.on('change', common.mustCallAtLeast(function(event, filename) {
+    // Libuv inconsistenly emits a rename event for the file we are watching
+    assert.ok(event === 'change' || event === 'rename');
```

Comment:
On the implementation you send `change` event, not `rename`. Why do we have a rename here? If this is the only thing we check in this pull request, `rename` should not be here. If this is required for macOS but not for `Linux`, we need to match the same exact behavior in different operating systems.

Reply:
the thing is `libuv` behaves inconsistently in this case

Reply:
it sometimes fires `rename` and sometimes `change`

Reply:
the inconsistency is in libuv, so it is out of the scope of the PR AFAIK

Reply:
On Linux, it's not Libuv that sends `change` event, but it's `recursive_watch.js`.

Reply:
This test runs on all supported platforms, right? Not just Linux?

Reply:
correct

Reply:
I'm referencing my previous comment on the change in `recursive_watch.js`. On macOS this event is `rename`, with the current change in `recursive_watch.js` which only effects Linux, it returns `change` event. This function should behave same in all platforms.